packages/ruleset/src/functions/operation-id-naming-convention.js

```diff
@@ -74,51 +82,87 @@ function operationIdNamingConvention(resolvedSpec) {
   return errors;
 }
 
+//
+// This function checks to make sure that the operation's operationId starts with
+// an acceptable verb (e.g. 'create', 'list', etc.) depending on various
+// characteristics of the operation's path and http method.
+//
+// This function should be kept in sync with the API Handbook guidance in this area,
+// located here: https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-operations
+//
+// The following table attempts to "codify" these rules so that the logic below is a little
+// easier to understand:
+//
+// HTTP   |                             | Does path end | # of param   | Acceptable   | Operation ID
+// Method | Path Example                | w/ param ref? | refs in path | Verbs        | Examples
+// -------|-----------------------------|---------------|--------------|--------------|---------------------------------
+// get    | /albums                     | N             | 0            | list         | list_albums
+//        | /albums/{id}                | Y             | 1            | get          | get_album
+//        | /albums/{id}/genres/{gid}   | Y             | 2+           | get, check   | get_album_genre, check_album_genre
+//        |                             |               |              |              |
+// post   | /albums                     | N             | 0            | create       | create_album
+//        |                             |               |              |              |
+// patch  | /albums/{id}                | Y             | 1            | update       | update_album
+//        |                             |               |              |              |
+// put    | /albums                     | N             | 0            | replace      | replace_albums
+//        | /albums/{id}                | Y             | 1            | replace      | replace_album
+//        | /albums/{id}/genre          | N             | 1            | replace, set | replace_album_genre, set_album_genre
+//        | /albums/{id}/producer/{pid} | Y             | 2+           | add          | add_album_producer
```

Comment:
```suggestion
//        | /albums/{id}/producer/{pid} | Y             | 2+           | replace, add | add_album_producer
```

I _think_ `replace` should always be a supported possibility for `PUT` (even in the final row here).